scripts-dev/check_newsfragment.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+#
+# A script which checks that an appropriate news file has been added on this
+# branch.
+#
+# As first argument, it requires the PR number, so that it can check that the
+# newsfragment has the correct name.
+#
+# Usage:
+#   ./scripts-dev/check_newsfragment.sh 382
+#
+# Exit codes:
+#   0: all is well
+#   1: the newsfragment is wrong in some way
+#   9: the script has not been invoked properly
+
+echo -e "+++ \033[32mChecking newsfragment\033[m"
```

Comment:
bit odd to mix `\033` and `\e` in the same file, but :man_shrugging:

Reply:
I nicked it from Synapse, but good spot. I'll use `\e` for both (TIL that there's `\e` for escape!) for the sake of doing the right thing :).